timesketch/lib/analyzers/contrib/hashlookup_analyzer.py

```diff
@@ -72,30 +73,59 @@ def mark_event(self, event, hash_value):
 
         Tags with validate emoji, adds a comment to the event.
         """
-        self.sketch.add_view(
-            view_name="Hashlookup",
-            analyzer_name=self.NAME,
-            query_string=('tag:"Hashlookup"'),
-        )
 
         event.add_comment(f"{self.hashlookup_url}sha256/{hash_value}")
 
         event.add_tags(["Hashlookup"])
         event.add_emojis([emojis.get_emoji("VALIDATE")])
         event.commit()
 
-    def query_hashlookup(self, query, hash_type):
+    def query_hashlookup(self, query, return_fields):
         """Get event from timesketch, request Hashlookup and mark event."""
-        events = self.event_stream(query_string=query, return_fields=[hash_type])
-        for event in events:
-            loc = event.source.get(hash_type)
 
-            if not loc in self.request_list:
-                result = self.get_hash_info(loc)
+        events = self.event_stream(query_string=query, return_fields=return_fields)
+        create_a_view = False
+        for event in events:
+            hash_value = None
+            for key in return_fields:
+                if key in event.source.keys():
+                    hash_value = event.source.get(key)
+                    break
+            if not hash_value:
+                error_hash_counter += 1
+                logger.warning(
+                    "No matching field with a hash found in this event! "
+                    "-- Event Source: %s",
+                    str(event.source),
+                )
+                continue
```

Comment:
With your adjusted `query_string` to use `_exists_`, do you still need this check?

Reply:
Fixed in 18325158858d824839fb5825bcdc6e1bd6c14552.